x-pack/solutions/security/plugins/security_solution/public/management/pages/event_filters/view/components/form.test.tsx

```diff
@@ -48,6 +48,39 @@ jest.mock('../../../../../common/hooks/use_license', () => {
   };
 });
 
+/** When some props and states change, `EventFilterForm` will recreate its internal `processChanged` function,
+ * and therefore will call it from a `useEffect` hook.
+ *
+ * Amongst the aforementioned props is the item itself, which comes from the outside - the test environment.
+ * Amongst the aforementioned states is the `hasFormChanged` state, which will change from `false` to `true`
+ * on the first user action.
+ *
+ * In the browser, the `item` prop is a state, a state from the parent component, therefore both
+ * `item` and `hasFormChange` will change at the same re-render, which ensures that the `useEffect` will
+ * call `processChanged` with consistent data.
+ *
+ * In the test environment, however, we need a trick to make sure that the data is updated in the same re-render
+ * both outside and inside the component, in order to not receive additional calls from `processChanged` with outdated
+ * data.
+ *
+ * This `TestComponentWrapper` component is meant to provide this kind of syncronisation, by turning the `item` prop
+ * into a state.
+ *
+ */
+const TestComponentWrapper: typeof EventFiltersForm = (formProps: ArtifactFormComponentProps) => {
```

Comment:
this feels definitely hacky, but looks like it's needed with the current architecture

i'll open a refactor PR to all artifact forms to try to solve the architectural issues